module1/homework/grayscaleImages/compression.cpp

```diff
@@ -0,0 +1,56 @@
+#include "compression.hpp"
+
+#include <iostream>
+
+std::vector<std::pair<uint8_t, uint8_t>> compressGrayscale(const std::array<std::array<uint8_t, width>, height>& bitmap) {
+    std::vector<std::pair<uint8_t, uint8_t>> compressed;
+    for (const auto& line : bitmap) {
+        uint8_t count{};
+        uint8_t pixel = line.at(0);
+        for (const auto& el : line) {
+            if (pixel == el) {
+                ++count;
+            } else {
+                compressed.push_back(std::make_pair(pixel, count));
+                count = 1;
+                pixel = el;
+            }
+        }
+        compressed.push_back(std::make_pair(pixel, count));
```

Comment:
Try to reconfigure your loop, to call push_back only once.

Reply:
Loop reconfigured; one call now